### PR TITLE
Swaps: subtract gas fee buffer from native token balances

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
* when input token is native token, remove a buffer of estimated gas fee * 1.3 to make sure the user has enough gas to power tx
* added `maxSwappableAmount` attribute to `ExtendedAnimatedAssetWithColors` - this value is initialized to asset balance and is updated to be estimated gas fee * 1.3 for native tokens

### CODE REVIEW

File: src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts [hunk 1]
```diff
@@ -39,6 +47,24 @@ export function useEstimatedGasFee({
...
+        const gasFeeNativeCurrency = divWorklet(totalWei, 10 ** asset.decimals);
+        const gasFeeWithBuffer = toFixedWorklet(mulWorklet(gasFeeNativeCurrency, 1.3), asset.decimals);
+        const maxSwappableAmount = subWorklet(asset.balance.amount, gasFeeWithBuffer);
+
+        return {
+          ...asset,
+          maxSwappableAmount: Number(maxSwappableAmount) < 0 ? '0' : maxSwappableAmount,
+        };
+      } else {
+        return asset;
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts:64]: should we set `maxSwappableAmount` to balance here?

File: src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts [hunk 2]
```diff
@@ -39,6 +47,24 @@ export function useEstimatedGasFee({
...
+      'worklet';
+
+      if (asset?.isNativeAsset) {
+        const gasFeeNativeCurrency = divWorklet(totalWei, 10 ** asset.decimals);
+        const gasFeeWithBuffer = toFixedWorklet(mulWorklet(gasFeeNativeCurrency, 1.3), asset.decimals);
+        const maxSwappableAmount = subWorklet(asset.balance.amount, gasFeeWithBuffer);
+
+        return {
+          ...asset,
+          maxSwappableAmount: Number(maxSwappableAmount) < 0 ? '0' : maxSwappableAmount,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts:61]: why not the `lessThanOrEqualToWorklet` here?

File: src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts [hunk 3]
```diff
@@ -39,6 +47,24 @@ export function useEstimatedGasFee({
     const amount = gasSettings.isEIP1559 ? add(gasSettings.maxBaseFee, gasSettings.maxPriorityFee) : gasSettings.gasPrice;
 
     const totalWei = multiply(gasLimit, amount);
+
+    internalSelectedInputAsset.modify(asset => {
+      'worklet';
+
+      if (asset?.isNativeAsset) {
+        const gasFeeNativeCurrency = divWorklet(totalWei, 10 ** asset.decimals);
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts:55]: leaving myself a note so I don't forget - not caused by this code but something I observe: sometimes I can get `gasLimit` back as `NaN` which messes up this `divWorklet`

Comment (brunobar79) [inline src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts:55]:
```suggestion
        const gasFeeNativeCurrency = divWorklet(totalWei, powWorklet(10, asset.decimals));
```

File: src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts [hunk 4]
```diff
@@ -39,6 +47,24 @@ export function useEstimatedGasFee({
     const amount = gasSettings.isEIP1559 ? add(gasSettings.maxBaseFee, gasSettings.maxPriorityFee) : gasSettings.gasPrice;
 
     const totalWei = multiply(gasLimit, amount);
+
+    internalSelectedInputAsset.modify(asset => {
+      'worklet';
+
+      if (asset?.isNativeAsset) {
+        const gasFeeNativeCurrency = divWorklet(totalWei, powWorklet(10, asset.decimals));
+        const gasFeeWithBuffer = toFixedWorklet(mulWorklet(gasFeeNativeCurrency, 1.3), asset.decimals);
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts:56]: i would define this buffer in the constants.ts file or something. 1.3 hardcoded here is a recipe for misunderstanding later on

**Review by walmat — APPROVED** [review]
LGTM other than Jin's comment

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**